src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -254,6 +254,37 @@ def get_inventories(hosts: Optional[List[str]] = None,
     return inventory_hosts
 
 
+def _check_orchestrator_host_op(orch_client, hostname, add_host=True):  # pragma:no cover
+    """Check if we can adding or removing a host with orchestrator
+
+    :param orch_client: Orchestrator client
+    :param add: True for adding host operation, False for removing host
+    :raise DashboardException
+    """
+    host = orch_client.hosts.get(hostname)
+    if add_host and host:
+        raise DashboardException(
+            code='orchestrator_add_existed_host',
+            msg='{} is already in orchestrator'.format(hostname),
+            component='orchestrator')
+    if not add_host and not host:
+        raise DashboardException(
+            code='orchestrator_remove_nonexistent_host',
+            msg='Remove a non-existent host {} from orchestrator'.format(hostname),
+            component='orchestrator')
+
+
+def create_host(hostname: str, addr: Optional[str] = None,
```

Comment:
`create_host` is a bit misleading name (I know it comes from the automatic mapping between `create` method and POST, but it's still confusing)... Are we really creating it or just adding it to the cluster (if that's the case, let's extend this method-to-endpoint mapper with more exact verbs)?

Reply:
I changed it to add_host(). Its more meaningful in this case.